src/auth/auth.service.ts

```diff
@@ -126,6 +128,93 @@ export class AuthService {
     }
   }
 
+  /**
+   * it create a user with a correct crendentials from the decoded token
+   * @param dto - object containing token
+   * @param values - an object containing userAgent and IpAddress of the user
+   * @returns {userSignInType} object containing information about the signin user
+   */
+  async googleSignin(
+    dto: GoogleSigninDto,
+    values: { userAgent: string; ipAddress: string },
+  ) {
+    try {
+      const client = new OAuth2Client(
+        await this.configService.get(configConstant.google.clientID),
+      );
+
+      const verifyClientToken = await client.verifyIdToken({
+        idToken: dto.token,
+        audience: await this.configService.get(configConstant.google.clientID),
+      });
+
+      const { name, email } = verifyClientToken.getPayload();
+
+      const user = await this.userRepo.findOne({ where: { email } });
+
+      if (!user) {
+        // crreate user account
+        const newUser = this.userRepo.create({
+          email,
+          fullName: name,
+          isGoogleAuthUser: true,
+        });
+        await this.userRepo.save(newUser);
```

Comment:
why dont u create a profile first and confirm it was created successfully before saving a user

Reply:
A user must be created before creating a profile